Add WeiKai component documentation

DrCoolzic opened the pull request:
Documentation for the new Weikai family of components

## Description:
This component adds support for the Weikai family of component chips. It add the following components:
wk2132_spi, wk2212_spi, wk2204_spi, wk2168_spi, wk2132_i2c, wk2212_i2c, wk2204_i2c, wk2168_i2c

The code for the component has been submitted with the PR #5218

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#5218
Add WK2132 component to esphome #5218
## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

DrCoolzic committed:
documentation ready to submit wk2132 to esphome-docs

I have created a new clean branch for releasing wk2132 documentation.
Note that as I do not submit the sc16is75x for now, I have removed
the table comparison.

On branch wk2132
Changes to be committed:
	new file:   components/images/DFR0627.jpg
	new file:   components/wk2132.rst
	new file:   images/wk2132.jpg
	modified:   index.rst

DrCoolzic committed:
Change sensor example to working one

seems that I have chosen an example that causes problems

On branch wk2132
Your branch is up to date with 'origin/wk2132'.

Changes to be committed:
	modified:   components/wk2132.rst

DrCoolzic committed:
change uart_id to uart_0

DrCoolzic committed:
Merge branch 'next' into wk2132

DrCoolzic committed:
fix lint error

DrCoolzic committed:
Merge branch 'wk2132' of https://github.com/DrCoolzic/esphome-docs into wk2132

DrCoolzic committed:
extra line

DrCoolzic committed:
Merge branch 'next' into wk2132

DrCoolzic committed:
validation has changed on ESPHome ???

DrCoolzic committed:
Merge branch 'wk2132' of https://github.com/DrCoolzic/esphome-docs into wk2132

DrCoolzic committed:
as per request?

DrCoolzic committed:
Merge branch 'esphome:current' into wk2132

DrCoolzic committed:
test

DrCoolzic committed:
remove test

DrCoolzic committed:
Merge branch 'esphome:current' into wk2132

DrCoolzic committed:
Merge branch 'next' into wk2132

DrCoolzic committed:
lint spaces for list-table

DrCoolzic committed:
Merge branch 'wk2132' of https://github.com/DrCoolzic/esphome-docs into wk2132

DrCoolzic committed:
Fixed lint & added performance information

On branch wk2132
Your branch is up to date with 'origin/wk2132'.

Changes to be committed:
	modified:   components/wk2132.rst

DrCoolzic committed:
Renamed component from wk2132 to wk2132_i2c

Also added information about performance, and transmission size ...

On branch wk2132
Your branch is up to date with 'origin/wk2132'.

Changes to be committed:
	renamed:    components/wk2132.rst -> components/wk2132_i2c.rst
	renamed:    images/wk2132.jpg -> images/wk2132_i2c.jpg
	modified:   index.rst

DrCoolzic committed:
Merge branch 'next' into wk2132

jesserockz committed:
Merge branch 'next' into wk2132

jesserockz committed:
Discard changes to components/esp32_improv.rst

jesserockz commented on components/wk2132_i2c.rst:
```diff
@@ -0,0 +1,179 @@
+chip or a DFROBOT `Gravity: I²C to Dual UART Module <https://www.dfrobot.com/product-2001.html>`__
+module connected to an ESP32 MCU through an I²C bus. 
+
+The WK2132 chip/module serves as an I²C to Dual UART converter, supporting data transmission rates 
+of up to 1 Mbps. While the chip can also interface with an SPI bus, this functionality is supported by 
+a separate ESPHome component (wk2132_spi). Each UART channel features two independent 256-byte FIFO hardware 
+buffers for transmission and reception.
```
Until such a time as that component exists, it is best not to document it as existing...
```suggestion
The WK2132 chip/module serves as an I²C to Dual UART converter, supporting data transmission rates 
of up to 1 Mbps. Each UART channel features two independent 256-byte FIFO hardware 
buffers for transmission and reception.
```

jesserockz commented on components/wk2132_i2c.rst:
```diff
@@ -0,0 +1,179 @@
+
+.. caution:: 
+
+    Currently, due to some internal limitations, it is necessary to declare a dummy ``uart:`` component in
+    your configuration **even if you do not need one**. See the example above. This requirement 
+    will disappear in future release.\ :raw-html-m2r:`<br>`
+
```
This limitation will be fixed before the component is merged.
```suggestion
```

jesserockz commented on components/wk2132_i2c.rst:
```diff
@@ -0,0 +1,179 @@
+********
+
+- :ref:`i2c`
+- :ghedit:`Edit`
```
```suggestion
- :apiref:`wk2132_i2c/wk2132_i2c.h`
- :ghedit:`Edit`
```

jesserockz commented on components/wk2132_i2c.rst:
```diff
@@ -0,0 +1,179 @@
+    maximum transmission size to 255. The current version of wk2132_i2c automatically sets this limit for 
+    you. This limitation will disappear in future versions
+
+Component configuration variables:
+**********************************
```
```suggestion
Configuration variables:
************************
```

DrCoolzic committed:
Removed all workarounds related to bugs

DrCoolzic committed:
Merge branch 'wk2132' of https://github.com/DrCoolzic/esphome-docs into wk2132

DrCoolzic committed:
Modified based on suggested changes

kbx81 commented on components/wk2132_i2c.rst:
```diff
@@ -0,0 +1,139 @@
+      sda: 21
+      scl: 22
+      frequency: 600kHz
+
```
Typically we don't include the configuration of dependent components in the example.
```suggestion
```

DrCoolzic committed:
Update components/wk2132_i2c.rst

Co-authored-by: Keith Burzinski <kbx81x@gmail.com>

DrCoolzic committed:
Merge branch 'esphome:current' into wk2132

DrCoolzic committed:
New documentation to cover wk2132_i2c & wk2132_spi

I would like to release the two versions on the same PR.
This make sense as the two components are deeply related
using a common wk2132 root component

On branch wk2132
Your branch is up to date with 'origin/wk2132'.

Changes to be committed:
	new file:   components/wk2132.rst
	deleted:    components/wk2132_i2c.rst
	renamed:    images/wk2132_i2c.jpg -> images/wk2132.jpg
	modified:   index.rst

DrCoolzic committed:
fix lint error

DrCoolzic committed:
Heading level

DrCoolzic committed:
changed .seo title

DrCoolzic commented:
Sorry as the PR was still open I took the oportunity to squeeze the SPI version as well :)

DrCoolzic committed:
Support WeiKai family of chips

This new version now supports the complete WeiKai family of UART
and GPIO Expanders

On branch wk2132
Your branch is up to date with 'origin/wk2132'.

Changes to be committed:
	new file:   components/weikai.rst
	deleted:    components/wk2132.rst
	deleted:    images/wk2132.jpg
	new file:   images/wk2168.jpg
	modified:   index.rst

DrCoolzic committed:
Reduce size of images

On branch wk2132
Your branch is up to date with 'origin/wk2132'.

Changes to be committed:
	modified:   images/wk2168.jpg

nagyrobi committed:
Merge branch 'next' into wk2132

DrCoolzic committed:
Fixed error in names and some mistakes

DrCoolzic committed:
fixed error in names ...

DrCoolzic committed:
Merge branch 'wk2132' of https://github.com/DrCoolzic/esphome-docs into wk2132

nagyrobi committed:
Merge branch 'next' into wk2132

nagyrobi commented:
@DrCoolzic can you please check if @jesserockz 's observations are still valid or not, and resolve them if necessarry? (if already resolved, mark them as resolved please)

DrCoolzic committed:
changed :aspiref:

DrCoolzic committed:
Merge branch 'wk2132' of https://github.com/DrCoolzic/esphome-docs into wk2132

DrCoolzic commented:
> @DrCoolzic can you please check if @jesserockz 's observations are still valid or not, and resolve them if necessarry? (if already resolved, mark them as resolved please)

I beleive I have now marked the request as resolved. I had to made a change to my rst file based on feedback from @jesserockz and I just pushed it few minutes ago

DrCoolzic committed:
Changed documentation to match request from Jesse

DrCoolzic commented:
Modified the documentation to reflect changes required by @jesserockz to uart_id
Plus miscellaneous spelling correction

nagyrobi committed:
Merge branch 'next' into wk2132

jesserockz committed:
Minor cleanups
Remove extra spaces

jesserockz committed:
Merge branch 'next' into pr/DrCoolzic/3113
